mmv1/templates/terraform/pre_delete/compute_network_endpoints.go.erb

```diff
@@ -0,0 +1,34 @@
+var endpointsToDelete []map[string]interface{}
```

Comment:
In general we find that having a list resource that partially manages things is pretty fragile and has unexpected behavior - I'd recommend making this an authoritative resource that deletes anything it doesn't manage. See iam binding behavior as another example: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_iam

It does have a risk of confusion from users but less so than the alternative. Still, could make sense to add some explicit documentation about it.

Reply:
I think this is actually already authoritative, though I wasn't explicitly trying to make it so. If I have 1 `google_compute_network_endpoints` and 1 `google_compute_network_endpoint` associated with the same NEG, and there is no overlap, repeated calls to `terraform apply` will alternate between `google_compute_network_endpoint` creating itself and `google_compute_network_endpoints` deleting the `google_compute_network_endpoint`. I think this is because the `Read` implementation for `google_compute_network_endpoints` pulls all endpoints, even the ones managed outside of the `google_compute_network_endpoints`, and then Terraform will delete the ones it doesn't recognize.

So I think the only thing to do here is to make it clear in the docs that this is authoritative for the NEG. I did cap the number of endpoints at 500 so I wouldn't have to worry about pagination, so I should probably uncap it and figure out how to paginate this.

Reply:
+1 we should only cap it if there is an API limitation

Reply:
In terms of implementing pagination - I was thinking we might have a standard way to handle this already, but it doesn't look like we do.. Given that - let's just cap it at 500 after all 😂

I think the alternative would be to use the custom decoder to pull the additional pages - but that will still be an option later if needed. Basically, I'd say it depends on how common / important of a use case you think having >500 endpoints would be.

Reply:
Using a custom decoder makes the most sense, I don't think there's a need to generally implement a paginated read.

I don't mind adding it to this PR, but I don't think it should be a PR blocker either.

Reply:
Turns out there is also a 500 item limit for `attachNetworkEndpoints` (and presumably `detachNetworkEndpoints`), so we would need to do pagination for create and delete as well. I'll leave that to a future PR if the need arises.